docs/source/traits_user_manual/notification.rst

```diff
@@ -719,3 +719,47 @@ steps other than 1, **index** holds the _slice_ that was changed.
 The TraitDictEvent has an additional **changed** attribute which holds the
 keys that were modified and the _old_ values that those keys held.  The new
 values can be queried from directly from the trait value, if needed).
+
+
+.. _on-trait-change-dos-n-donts:
+
+
+Dos and Don’ts
+--------------
+
+Do not assume handlers are called in a specific order
+`````````````````````````````````````````````````````
+Don't do this::
+
+    @on_trait_change("name")
+    def update_number(self):
+        self.number += 1
+
+    @on_trait_change("name")
+    def update_orders(self):
+        if self.number > 5:
+          self.orders.clear()
+
+Do this instead::
+
+    @on_trait_change("name")
+    def update(self):
+        number = self.number + 1
+        self.number = number
+        if number > 5:
+            self.orders.clear()
+
+The first example is problematic because when ``name`` changes, calling
+``update_orders`` after ``update_number``  produces a result that is different
+from calling ``update_number`` after ``update_orders``.
+
+Even if the change handlers appear to be called in a deterministic order,
+this would be due to implementation details that may not hold true across
+releases and platforms.
+
+Traits consider handlers for the same change event to be independent of each
```

Comment:
Do you think we can break this paragraph (or part of it) out into a separate "don't", namely: "don't raise exceptions from change handlers"? Or maybe that's better done in a separate PR?

Reply:
Also, it's kinda obvious, but "don't return values from change handlers" might be worth having, too.

Reply:
I imagine this section to carry on growing 🌴.  Yes I can separate these two out here.

Reply:
Actually, I ended up not adding "don't return values from change handlers".  It is unlikely anyone has been surprised by it so it might end up clustering the section.